Remove Clair from docs because it is deprecated in 2.2

a-mccarthy opened the pull request:
fixes #175 and #174

a-mccarthy commented:
Setting as draft for now. Still need to add information around how upgrade works for folks using clair as the default scanner before upgrade

xaleeks commented on docs/administration/upgrade/upgrade-test.md:
```diff
@@ -7,15 +7,15 @@ weight: 50
 2. Create project projecta projectc as private, create projectb as public.  
 3. Add usera as projecta's admin, userc as developer, and userd as guest. Do the same to projectb and projectc.  
-4. Login harbor as usera, push an unsigned image into projecta, then push a signed image to projecta. 
+4. Login harbor as usera, push an unsigned image into projecta, then push a signed image to projecta.
```
user a

a-mccarthy replied:
usera is the name of the users from the instructions above.

xaleeks commented on docs/administration/vulnerability-scanning/_index.md:
```diff
@@ -3,30 +3,17 @@ title: Vulnerability Scanning
 
-It might be necessary to connect Harbor to other scanners for corporate compliance reasons, or because your organization already uses a particular scanner. Different scanners also use different vulnerability databases, capture different CVE sets, and apply different severity thresholds. By connecting Harbor to more than one vulnerability scanner, you broaden the scope of your protection against v ...
+{{< important >}}
+Clair was removed as a default scanner in V2.2. You are still able to use Clair for vulnerability scanning by adding it as additional scanner.
```
i'd change it to 'by adding it as an external scanner'

a-mccarthy replied:
updated

xaleeks commented on docs/administration/vulnerability-scanning/import-vulnerability-data.md:
```diff
@@ -4,7 +4,7 @@ weight: 45
 
 {{< important >}}
-Clair is being removed as a default vulnerability scanner in Harbor v2.2. Its highly recommended that you configure Trivy as your default scanner instead. If you want to continue using Clair, you should configure it as an [external scanner](../../vulnerability-scanning/pluggable-scanners/).
+Clair was removed as the default vulnerability scanner in Harbor V2.2. Its highly recommended that you configure Trivy as your default scanner instead. If you want to continue using Clair in V2.2 and later, you must configure it as an [additional scanner](../../vulnerability-scanning/pluggable-scanners/).
```
Clair has been removed ?

xaleeks replied:
-> 'It's highly recommended'

xaleeks replied:
V2.2 -> v2.2

a-mccarthy replied:
updated

xaleeks commented on docs/install-config/configure-yml-file.md:
```diff
@@ -245,7 +240,7 @@ The following table lists the additional, optional parameters that you can set t
     <td valign="top"><code>external_database</code></td>
     <td valign="top">&nbsp;</td>
-    <td valign="top">Configure external database settings, if you disable the local database option. Currently, Harbor only supports PostgreSQL database. You must create four databases for Harbor core, Clair, Notary server, and Notary signer. The tables are generated automatically when Harbor starts up.</td>
+    <td valign="top">Configure external database settings, if you disable the local database option. Currently, Harbor only supports PostgreSQL database. You must create three databases for Harbor core, Notary server, and Notary signer. The tables are generated automatically when Harbor starts up.</td>
```
three separate databases? or tables?

a-mccarthy replied:
i assumed databases, since that is what is said before. should it be tables? Re-reading this, it makes more sense to say databases here, since the next section says "tables are generated automatically when Harbor starts up", why would you need to create the 3 tables if they are generated automatically when Harbor starts?


Would a user still need to create let harbor know about an external Clair database to use Clair as an external scanner?

xaleeks replied:
lgtm

xaleeks commented on docs/install-config/reconfigure-manage-lifecycle.md:
````diff
@@ -63,10 +63,10 @@ To reconfigure Harbor, perform the following steps.
 
     ```sh
-    sudo prepare --with-notary --with-clair --with-chartmuseum
+    sudo prepare --with-notary --with-chartmuseum
````
wait..do we not have an option to install with trivy?

a-mccarthy replied:
i updated this to say --with-trivy

a-mccarthy commented:
@xaleeks i updated this PR from your review comments and I added upgrade data. 

Is there an issue to update the rest of the Upgrade page, https://goharbor.io/docs/edge/administration/upgrade/, for 2.2?

sluetze commented on docs/administration/upgrade/_index.md:
```diff
@@ -13,8 +13,9 @@ Since the migration might alter the database schema and the settings of `harbor.
 
 - Again, you MUST backup your data before any data migration.
-- In version 1.10.0, some containers are started by `non-root`. This does not pose problems if you are upgrading an officially released version of Harbor, but if you have deployed a customized instance of Harbor, you might encounter permission issues.
+- In version 1.10.0, some containers are started by `non-root`. This does not pose problems if you are upgrading an ially released version of Harbor, but if you have deployed a customized instance of Harbor, you might encounter permission issues.
```
typo?

a-mccarthy replied:
oh yes! weird, i dont know how that happened. thanks for raising!

a-mccarthy replied:
should be fixed now

a-mccarthy commented:
@xaleeks can you also provide your thoughts on what we should do with the quick install script? Right now it uses clair by default but i no longer being maintained, see #159.

Id recommend that we remove the script OR plan to maintain it as part of the repo, do we have a place that we normally place sample scripts?

a-mccarthy commented:
Updated the PR to remove the Quick install script. It was decided that this should no longer be part of the docs since it is no longer maintained. 

closes #159

heww commented:
https://github.com/goharbor/website/blob/master/docs/administration/vulnerability-scanning/import-vulnerability-data.md This file may need to be removed.

a-mccarthy commented:
>https://github.com/goharbor/website/blob/master/docs/administration/vulnerability-scanning/import-vulnerability-data.md This file may need to be removed.

I asked @stonezdj about this, and he confirmed this is still something a user can do with Clair added as an external scanner.

xaleeks commented:
@heww can you check again and merge this if it's good?
